src/RpcClient.ts

```diff
@@ -631,7 +640,11 @@ export class RpcClient {
 
           return await this.pollTransactionConfirmation(txtSig);
         } catch (error) {
-          continue;
+          if (error instanceof TransactionExpiredTimeoutError) {
+            continue;
+          } else {
+            throw error;
```

Comment:
@0xIchigo I think we should throw custom error messages if it's not an expired timeout one. What do you think?

Reply:
I like this, but for now, I think we should just continue

Reply:
I have to do some work for `sendSmartTransaction` so I'll merge as is, instead of waiting for this to be changed, and I'll update it myself